flang/include/flang/Optimizer/Dialect/FIROps.td

```diff
@@ -3203,6 +3205,9 @@ def fir_DeclareOp : fir_Op<"declare", [AttrSizedOperandSegments,
     It must always be provided for characters and parametrized derived types
     when memref is not a box value or address.
 
+    The storage and storage_offset operands are optional and are required
+    for FortranVariableStorageOpInterface, where they are documented.
+
```

Comment:
It would be good to mention this for the HLFIR op too, and to explain the defaults assumed when these new attributes are not specified.